Muzima 825

saraheisa opened the pull request.

saraheisa committed:
MUZIMA-825

Fix: edit the openmrs supported versions to include the latest version

saraheisa committed:
MUZIMA-825

Feat: Require the htmlFormEntry module

Also need to run the script here to make it work:
https://wiki.openmrs.org/display/docs/Requiring+another+module+in+your+module

The module is needed to import the htmlforms

saraheisa committed:
MUZIMA-825

Feat: handle the requests from htmlFormEntry import

handles two requests:
1) get the htmlforms from the htmlFormEntryService
2) check the htmlFormEntry module status

Created convertHtmlForm method in WebConverter to generate the list of
htmlforms for display

saraheisa committed:
MUZIMA-825

Feat: add Alert library

Alert is used to show notifications for the user

saraheisa committed:
MUZIMA-825

Feat: add HtmlFormEntryService for htmlFormEntry import

communicate with the server to:
1) check the htmlFormEntry module status
2) get htmlforms
3) TODO: convert the htmlForm
4) TODO: save muzima form
5) get getDiscriminatorTypes

saraheisa committed:
MUZIMA-825

Feat: add htmlFormEntry Component

1) add routing path for htmlFormEntry
2) add html/controller

In the controller I had to add the modal controllers their because
it didn't work otherwise

saraheisa committed:
MUZIMA-825

Feat: add import htmlformentry option

saraheisa committed:
MUZIMA-825

Feat: add muzima.css file for muzima form preview in htmlFormEntry import

saraheisa commented:
@mssavai @benamonya 
Please review the PR
Thanks

saraheisa committed:
MUZIMA-825

Feat: create a method to handle the convert request

the method returns a dummy response for now

saraheisa committed:
MUZIMA-825

Feat: replace the dummy response by a convert request

delete the save method because it's redundant since we're using
FormService to save the forms

saraheisa committed:
MUZIMA-825

Feat: change the save functionality to use FormService instead of
HtmlFormEntryService

needed to send the FormService object to the form modal so the
formModalInstance uses it when to save the form

saraheisa committed:
MUZIMA-825

FIX: Amend htmlformentry dependency to have htmlformentry-api

need to add the -api to locate the jar file via maven
with the -api included you don't have to run any scripts

benamonya commented on omod/src/main/webapp/resources/js/custom/app.js:
```diff
@@ -220,6 +221,35 @@ muzimaCoreModule.factory('FileUploadService', function ($http) {
+
+    var moduleState = function () {
+
+        return $http.get('htmlFormEntrymoduleStatus.form');
```
For consistency purposes consider naming this using camelCase. m Should be capital. Likewise htmlformentry.form should be in camelCase

benamonya commented on omod/src/main/webapp/resources/js/custom/controllers/HtmlFormEntryController.js:
```diff
@@ -0,0 +1,244 @@
+                console.info(error);
+            });
+
+        // HtmlFormEntryService.getDiscriminatorTypes()
```
Remove the commented out code

benamonya commented on omod/src/main/webapp/resources/js/custom/controllers/HtmlFormEntryController.js:
```diff
@@ -0,0 +1,244 @@
+    });
+
+    // prevent routing while converting
+
```
Remove spaces to enable one know it is a comment for the function

benamonya commented on omod/src/main/webapp/resources/partials/htmlFormEntry.html:
```diff
@@ -0,0 +1,278 @@
+                </div>
+            </div>
+
+            <!-- <div class="row panel panel-default" ng-show="totalItems > 1">
```
Remove commented out code

benamonya commented on omod/src/main/webapp/resources/partials/htmlFormEntry.html:
```diff
@@ -0,0 +1,278 @@
+                                <div class="form-review">
+                                    <div ng-show="$ctrl.converting" class="wait-convert"></div>
+                                    <uib-tabset active="active" ng-show="!$ctrl.converting">
+                                        <uib-tab index="0" heading="HTML Review" class="tab-review" style="width: 50%; text-align: center !important; color: #004f47 !important;">
```
Review should be renamed to Preview

saraheisa committed:
MUZIMA-825

Fix: rename HtmlFormEntryController urls using camelCase
for consistency purposes

saraheisa committed:
MUZIMA-825

Style: remove commented out code

saraheisa committed:
MUZIMA-825

Style: remove spaces after comments to enable one know it's a comment
for the function

saraheisa committed:
MUZIMA-825

Refactor: rename review to preview
